R/serializer.R

```diff
@@ -146,56 +156,72 @@ serializer_content_type <- function(type, serialize_fn = identity) {
     stop("You must provide the custom content type to the serializer_content_type")
   }
 
+  stopifnot(length(type) == 1)
+  stopifnot(is.character(type))
+  stopifnot(nchar(type) > 0)
+
   serializer_headers(
     list("Content-Type" = type),
     serialize_fn
   )
 }
 
-#' @describeIn serializers CSV serializer. See \code{\link[readr:format_delim]{readr::format_csv()}} for more details.
+#' @describeIn serializers CSV serializer. See also: \code{\link[readr:format_delim]{readr::format_csv()}}
 #' @export
-serializer_csv <- function(...) {
+serializer_csv <- function(..., type = "text/csv; charset=UTF-8") {
   if (!requireNamespace("readr", quietly = TRUE)) {
     stop("`readr` must be installed for `serializer_csv` to work")
   }
 
-  serializer_content_type("text/csv; charset=UTF-8", function(val) {
+  serializer_content_type(type, function(val) {
     readr::format_csv(val, ...)
   })
 }
 
+#' @describeIn serializers TSV serializer. See also: \code{\link[readr:format_delim]{readr::format_tsv()}}
```

Comment:
Do this for csv as well

```suggestion
#' @describeIn serializers TSV serializer. See also: [readr::format_tsv()]
```